storage/s3.go

```diff
@@ -330,10 +330,20 @@ func (s *S3) listObjectsV2(ctx context.Context, url *url.URL) <-chan *Object {
 
 				objectFound = true
 			}
+
 			// track the instant object iteration began,
 			// so it can be used to bypass objects created after this instant
 			if now.IsZero() {
-				now = time.Now().UTC()
+				if serverDate != "" {
+					n, err := http.ParseTime(serverDate)
```

Comment:
`http.ParseTime` function is not able to accurately represent the last modified time of an S3 object, because the latter has a much higher precision. This is causing the tests to fail. 

For example:
- Parsed time in header : 2023-07-01 19:43:58 +0000 UTC
- Last Modified time of S3 object: 2023-07-01 19:43:58.058 +0000 UTC

I will let you know if I am able to find a solution to the time [comparison](https://github.com/peak/s5cmd/pull/567/files#diff-22a91901acea81cf09b58954babe267e1a5b503dbce7c37ab0df11ebfc86b0b1R355) issue.